src/main/java/it/gov/pagopa/pu/debtpositions/service/DebtPositionServiceImpl.java

```
@@ -93,5 +97,96 @@ public PagedDebtPositions getPagedDebtPositionsByIngestionFlowFileId(Long ingest
+        if (poEntity != null) {
+          propagateIdsForPaymentOption(poEntity, poDTO);
+        }
+      });
+  }
+
+  private void propagateIdsForPaymentOption(PaymentOption poEntity, PaymentOptionDTO poDTO) {
+    if (poDTO.getPaymentOptionId() == null && poEntity.getPaymentOptionId() != null) {
+      poDTO.setPaymentOptionId(poEntity.getPaymentOptionId());
+    }
+
+    Map<String, InstallmentNoPII> instByIud = indexBy(
+      poEntity.getInstallments(),
+      InstallmentNoPII::getIud
+    );
+
+    streamOf(poDTO.getInstallments())
+      .forEach(instDTO -> {
+        InstallmentNoPII instEntity = instByIud.get(instDTO.getIud());
+        if (instEntity != null) {
+          propagateIdsForInstallment(instEntity, instDTO);
+        }
+      });
+  }
+
+  private void propagateIdsForInstallment(InstallmentNoPII instEntity, InstallmentDTO instDTO) {
+    if (instDTO.getInstallmentId() == null && instEntity.getInstallmentId() != null) {
+      instDTO.setInstallmentId(instEntity.getInstallmentId());
+    }
+
+    Map<Integer, Transfer> trByIndex = indexBy(
+      instEntity.getTransfers(),
+      Transfer::getTransferIndex
+    );
+
+    streamOf(instDTO.getTransfers())
+      .forEach(trDTO -> {
+        Transfer trEntity = trByIndex.get(trDTO.getTransferIndex());
+        if (trEntity != null && trDTO.getTransferId() == null) {
+          trDTO.setTransferId(trEntity.getTransferId());
+        }
+      });
+  }
+
+  private static <T> Stream<T> streamOf(Collection<T> c) {
+    return c == null ? Stream.empty() : c.stream().filter(Objects::nonNull);
+  }
+
+  private static <T, K> Map<K, T> indexBy(Collection<T> items, Function<T, K> keyFn) {
+    return streamOf(items)
+      .map(t -> new AbstractMap.SimpleEntry<>(keyFn.apply(t), t))
+      .filter(e -> e.getKey() != null)
+      .collect(Collectors.toMap(
+        Map.Entry::getKey,
+        Map.Entry::getValue,
+        (a, b) -> a
+      ));
```

if key value is we should not ignore! but let the code throw an error!
It should return also an error in case of duplicates...
Anyway, you are grouping the entity fetch from DB, we should avoid to check the null value, cause it cannot happen
```suggestion
    return streamOf(items)
      .collect(Collectors.toMap(
        keyFn,
        Function.identity()
      ));
```